src/runtime/engine/flow_node_handler/receive_task_handler.ts

```
@@ -0,0 +1,95 @@
+import {IEventAggregator, ISubscription} from '@essential-projects/event_aggregator_contracts';
+import {IIdentity} from '@essential-projects/iam_contracts';
+
+import {ILoggingApi} from '@process-engine/logging_api_contracts';
+import {IMetricsApi} from '@process-engine/metrics_api_contracts';
+import {
+  eventAggregatorSettings,
+  IFlowNodeInstanceService,
+  IProcessModelFacade,
+  IProcessTokenFacade,
+  MessageEventReachedMessage,
+  Model,
+  NextFlowNodeInfo,
+  Runtime,
+} from '@process-engine/process_engine_contracts';
+
+import {FlowNodeHandler} from './index';
+
+export class ReceiveTaskHandler extends FlowNodeHandler<Model.Activities.ReceiveTask> {
+  private _eventAggregator: IEventAggregator;
+
+  constructor(eventAggregator: IEventAggregator,
+              flowNodeInstanceService: IFlowNodeInstanceService,
+              loggingService: ILoggingApi,
+              metricsService: IMetricsApi) {
+    super(flowNodeInstanceService, loggingService, metricsService);
+    this._eventAggregator = eventAggregator;
+  }
+
+  private get eventAggregator(): IEventAggregator {
+    return this._eventAggregator;
+  }
+
+  protected async executeInternally(receiveTaskActivity: Model.Activities.SendTask,
+                                    token: Runtime.Types.ProcessToken,
+                                    processTokenFacade: IProcessTokenFacade,
+                                    processModelFacade: IProcessModelFacade,
+                                    identity: IIdentity): Promise<NextFlowNodeInfo> {
+
+    await this.persistOnEnter(receiveTaskActivity, token);
+
+    const noMessageDefinitionProvided: boolean = receiveTaskActivity.messageEventDefinition === undefined;
+    if (noMessageDefinitionProvided) {
+      throw new Error('SendTask has no MessageDefinition!');
```

See above, should be a 422 `UnprocessableEntityError`.

---

Since we throw normal errors in the other handlers, I thought that it will be better to also throw a normal error here (consistency wise).